### PR TITLE
[stdlib] Various revisions and fixes for documentation

### PR DESCRIPTION
**Pull request by natecook1000**
- Fix wording for RandomAccessCollection
- Add note about array growth to reserveCapacity(_:)
- Reformat lazy flatMap discussions
- Improve Collection symbol consistency

### CODE REVIEW

**Comment by natecook1000** [comment]
@swift-ci Please smoke test

File: stdlib/public/core/Arrays.swift.gyb
Comment (stephentyrone) [inline stdlib/public/core/Arrays.swift.gyb]: Style question: "lower" or "worse"?

File: stdlib/public/core/Arrays.swift.gyb
Comment (stephentyrone) [inline stdlib/public/core/Arrays.swift.gyb]: Do we have a standard method of marking code snippets that are showing "what not to do"? It would be nice to be explicit for someone skimming the text with a `// don't do this`, etc.

**Review by stephentyrone — COMMENTED** [review]
Looks good, except for two small style questions inline. I have only reviewed Arrays.swift.gyb, as the other stuff is outside my sphere of expertise.

File: stdlib/public/core/Arrays.swift.gyb
Comment (natecook1000) [inline stdlib/public/core/Arrays.swift.gyb]: "worse" is certainly clearer

File: stdlib/public/core/Arrays.swift.gyb
Comment (natecook1000) [inline stdlib/public/core/Arrays.swift.gyb]: We don't really—there are cases where we mark that something won't compile, but haven't had cases quite like this. I added a comment.

**Comment by stephentyrone** [comment]
Thanks Nate.  I'm happy with the Array section.

@airspeedswift please go ahead and approve once you review, if you like.

File: stdlib/public/core/Arrays.swift.gyb
Comment (airspeedswift) [inline stdlib/public/core/Arrays.swift.gyb]: an -> a

File: stdlib/public/core/RandomAccessCollection.swift
Comment (Gankro) [inline stdlib/public/core/RandomAccessCollection.swift]: I honestly preferred the old wording. Is this an actual rule we have?

File: stdlib/public/core/RandomAccessCollection.swift
Comment (Gankro) [inline stdlib/public/core/RandomAccessCollection.swift]: I'll admit that the above line is 100% more clear than this goop, but shouldn't the line above be listed as a "precondition"?

**Review by Gankra — COMMENTED** [review]
Meta: is there any coherent rule for when docs should be duplicated from the "declaration" of a method and its various implementations in extensions?

File: stdlib/public/core/Arrays.swift.gyb
Comment (natecook1000) [inline stdlib/public/core/Arrays.swift.gyb]:
```
- an
+ a
```

File: stdlib/public/core/FlatMap.swift
Comment (natecook1000) [inline stdlib/public/core/FlatMap.swift]: I'm only using "concatenated" when the result concatenates sequences/collections, not the optional unwrapping `flatMap`

**Comment by natecook1000** [comment]
@Gankro Thanks for the `flatMap` feedback! I'm working on more comprehensive changes there since it shows up in a variety of places other than just the lazy collections. I'd like to merge this now essentially as a bug fix for the dangling "Equivalent to" in the discussion—I'll let you know when I have revised documentation for the rest.

**Comment by natecook1000** [comment]
@swift-ci Please smoke test

File: stdlib/public/core/Arrays.swift.gyb
Comment (stephentyrone) [inline stdlib/public/core/Arrays.swift.gyb]: "to achieve" seems correct to me.

File: stdlib/public/core/RandomAccessCollection.swift
Comment (xwu) [inline stdlib/public/core/RandomAccessCollection.swift]: Is there a reason for the elimination of this discussion?

Comment (natecook1000) [inline stdlib/public/core/RandomAccessCollection.swift]: Unlike other `Indices` types, a `CountableRange<_>` doesn't hold a reference to the collection, so that part of the discussion isn't accurate. It still shows up on the protocol requirement and the unconstrained `indices` implementation.

**Review by airspeedswift — APPROVED** [review]

**Comment by natecook1000** [comment]
@swift-ci Please smoke test and merge

**Comment by natecook1000** [comment]
@swift-ci Please smoke test and merge

**Pull request merged**
